colorama/winterm.py

```diff
@@ -89,31 +91,38 @@ def get_position(self, handle):
         position.Y += 1
         return position
 
+    def get_cursor_position(self, on_stderr=False, adjust=True):
+        handle = self.get_handle(on_stderr)
+        info = win32.GetConsoleScreenBufferInfo(handle)
+        position = info.dwCursorPosition
+        # Because Windows coordinates are 0-based,
+        # and win32.SetConsoleCursorPosition expects 1-based.
+        y, x = position.Y + 1, position.X + 1
+        if adjust:
+            window = info.srWindow
+            y -= window.Top
+            x -= window.Left
+        return y, x
+
     def set_cursor_position(self, position=None, on_stderr=False):
         if position is None:
             # I'm not currently tracking the position, so there is no default.
             # position = self.get_position()
             return
-        handle = win32.STDOUT
-        if on_stderr:
-            handle = win32.STDERR
+        handle = self.get_handle(on_stderr)
         win32.SetConsoleCursorPosition(handle, position)
 
     def cursor_adjust(self, x, y, on_stderr=False):
-        handle = win32.STDOUT
-        if on_stderr:
-            handle = win32.STDERR
-        position = self.get_position(handle)
-        adjusted_position = (position.Y + y, position.X + x)
+        (cy, cx) = self.get_cursor_position(on_stderr, adjust=False)
```

Comment:
If we decide to continue to use get_position instead of the new get_cursor_position, we will not need this change.